### ISSUE TITLE
[kernel] On some systems, `/boot` is not mounted before install phase

### ISSUE THREAD

**Issue by cuantar** [macaroni-os/mark-issues#46]
This can probably be done with `mount-boot.eclass` that is in our tree.  The effect of not doing it is that, on systems with `/boot` on a separate partition (many/most?) the kernel and initramfs will not be where they should be, and won't be in the GRUB list created by `boot-update`.

Workaround:  Mount `/boot` manually (by _e.g._ `$ sudo mount /boot`) before emerging `debian-sources`.

So far it is unclear under what circumstances `/boot` is not mounted.  Currently experimenting to determine when it is not, and why not.

**Comment by cuantar** [comment]
I believe I have found the source of the problem:  The `mount-boot.eclass` has a regex that is too permissive.

The original:
```
grep -E "/boot" /etc/fstab | awk '{ print $1 }'
```

My replacement:
```
grep -E "^/.*[[:space:]]/boot[[:space:]]" /etc/fstab | awk '{ print $1 }'
```

Try executing both of them on a system with a comment in `/etc` that contains the string `/boot`, or on a system with `/boot/efi` defined.  In either case, the output of the original will not be what is expected, and `/boot` won't get mounted by `emerge`.

**Comment by cepelinas9000** [comment]
There is one more case, when filesystems are definied by uuid:
```
UUID=712727ce-0493-4718-ac11-e947c4da2deb     /boot               ext4        noatime      1 2
UUID=B995-2FF0    /boot/efi            vfat        noatime      1 2

```
the regex can look like this
```bash
grep -E "^(/|UUID=).*[[:space:]]/boot[[:space:]]" /etc/fstab | awk '{ print $1 }'
```

**Comment by cuantar** [comment]
Fixed in macaroni-os/kit-fixups#60.
